src/scheduler.service.ts

```diff
@@ -0,0 +1,25 @@
+import { scheduleJob as schedule } from 'node-schedule';
+import { log } from './logger';
+
+/**
+ * Schedule the given job function to run in the periodicity specified with cronTime
+ * @param timeRule cron expression that specifies when the job is ran or the date timestamp or a Date instance.
+ * Check this link to know the format: https://www.npmjs.com/package/node-schedule#cron-style-scheduling
+ * @param job callback function
+ * @param jobName (optional) name for the job
+ */
+export function scheduleJob(timeRule: string | number | Date, job: () => Promise<void>, jobName?: string) {
+  jobName = jobName ? ` ${jobName}` : '';
+  log.info(`Scheduling the job${jobName}`);
+
+  schedule(timeRule, async () => {
```

Comment:
This is not important for now, just some food for thought for the future 🤓 - should we store in the DB and load from the DB, because the node instance could be replace with another or load balanced for example.

Reply:
hmmm I haven't thought about this, this is indeed a problem! If the node instance goes down, the user won't receive the reminder. So yes, storing on the DB is a good solution for this.

I'd suggest a new collection called `jobs` that have a `id`, `reminderMessage`, `status` that is like an enum with the values "scheduled", "running", "finished", "error", and an optional `name`. Then somewhere in the startup of the bot, we read from it to reschedule the jobs that are in an error state?? This would also require to catch the shutdown of the node process and do some graceful shutdown.

Let me know your thoughts on implementing this.

Reply:
Sounds good. Does it need a date field too? Btw this feature can come in another PR, we can get this out and some feedback sooner

Reply:
I created an issue for this: #101 @eddiejaoude. I think it's better to keep it on another PR and keep the discussion for it's implementation there.
What do you think?

Reply:
Sounds like a good plan 👍